monitor.py

```diff
@@ -2157,6 +2194,23 @@ def suit_loadout_slots_array_to_dict(self, loadout: dict) -> dict:
 
         return slots
 
+    def _parse_navroute_file(self) -> dict[str, Any] | None:
+        """Read and parse NavRoute.json."""
+        if self.currentdir is None:
+            raise ValueError('currentdir unset')
+
+        with open(join(self.currentdir, 'NavRoute.json'), 'r') as f:
+            raw = f.read()
```

Comment:
If we're being paranoid, then we ought to catch things like FileNotFound in this open().

The function signature very much implies we'll always get `None` back for any error, not an exception thrown (which as far as I can see won't get caught in code anywhere near to the call point).

Reply:
Sure, there's a top-level try/except in EDLogs.parse_entry(), but that's going to imply the journal entry was bad, not that the additional file caused an issue.

Reply:
I've added an exists check but this can still technically race. Would you rather I wrap it anyway?

Reply:
I'd wrap it, as that means no race *in that code section*, in terms of "is the file there and we can open it?".  Whether data is yet available to be read is a whole other racey matter.